annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -507,6 +510,25 @@ private String resolveConsequence() {
         }
         return "";
     }
+    
+    private String resolveEntrezGeneId(boolean replace) {
+        if (!replace) {
+            return mRecord.getEntrez_Gene_Id();
+        }
+        // make the rest call to ensembl server for gene external refs
+        RestTemplate restTemplate = new RestTemplate();
+        HttpEntity<LinkedMultiValueMap<String, Object>> requestEntity = getRequestEntity();        
+        ResponseEntity<GeneXref[]> responseEntity = restTemplate.exchange(geneXrefsServiceUrl + canonicalTranscript.getGeneId(), HttpMethod.GET, requestEntity, GeneXref[].class);
```

Comment:
We also can only get this gene external reference info by gene id. We would need much more info to do the same thing by transcript id, which takes into consideration the CDS coordinates and chromosome, if I recall correctly